core/src/wsv.rs

```diff
@@ -279,23 +279,155 @@ impl WorldStateView {
     /// you likely have data corruption.
     /// - If trigger execution fails
     /// - If timestamp conversion to `u64` fails
-    #[iroha_futures::telemetry_future]
-    #[log(skip(self, block))]
     #[allow(clippy::expect_used)]
-    pub async fn apply(&self, block: VersionedCommittedBlock) -> Result<()> {
+    pub fn apply(&mut self, block: VersionedCommittedBlock) -> Result<()> {
```

Comment:
Because you can't call async code from sync code without a runtime. For more stability now and more performance later new-sumeragi is manually threaded. This means that it runs outside any tokio runtime and all functions it calls has to be sync.